test/unit/internal/oidc/test_ambient.py

```diff
@@ -153,6 +168,32 @@ def test_detect_github(monkeypatch):
             "fakeurl",
             params={"audience": "some-audience"},
             headers={"Authorization": "bearer faketoken"},
+            timeout=DEFAULT_TIMEOUT,
+        )
+    ]
+    assert resp.json.calls == [pretend.call()]
+
+
+def test_detect_github_timeout(monkeypatch):
+    # Check that the configured timeout is used for all `requests` calls.
+    monkeypatch.setenv("GITHUB_ACTIONS", "true")
+    monkeypatch.setenv("ACTIONS_ID_TOKEN_REQUEST_TOKEN", "faketoken")
+    monkeypatch.setenv("ACTIONS_ID_TOKEN_REQUEST_URL", "fakeurl")
+
+    resp = pretend.stub(
+        raise_for_status=lambda: None,
+        json=pretend.call_recorder(lambda: {"value": "fakejwt"}),
+    )
+    requests = pretend.stub(get=pretend.call_recorder(lambda url, **kw: resp))
+    monkeypatch.setattr(ambient, "requests", requests)
+
+    assert ambient.detect_github("some-audience", 0.1) == "fakejwt"
+    assert requests.get.calls == [
+        pretend.call(
+            "fakeurl",
+            params={"audience": "some-audience"},
+            headers={"Authorization": "bearer faketoken"},
+            timeout=0.1,
```

Comment:
I'm just checking that when we pass in a parameter for `timeout`, that it gets passed to all `requests` calls. If that seems like overkill, I can revert these.

Reply:
Works for me!